### DISCUSSION TITLE
Can ADGuardHome run independently without relying on Windows logs?

### DISCUSSION THREAD

**Discussion by SuXiChao1101** [AdguardTeam/AdGuardHome#6218] [Q&A] [answered]
I used some scripts that prohibit Windows from generating logs, such as the following:

`
set "key=HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\WINEVT\Channels"
for /f "delims=" %%A in ('reg query %key%') do (
      echo "%%A"
      reg add "%%A" /v "Enabled" /t REG_DWORD /d 0 /f
)

echo off

set "key=HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\WMI\Autologger\EventLog-System"
for /f "delims=" %%A in ('reg query %key%') do (
      echo "%%A"
      reg add "%%A" /v "Enabled" /t REG_DWORD /d 0 /f
      reg add "%%A" /v "EnableProperty" /t REG_DWORD /d 0 /f
)

`

Afterwards, ADGuardHome seemed to be unable to function, and starting ADGuardHome in a Windows service would prompt 'Error 1067'. Is there any way to enable ADGuardHome to start independently without relying on Windows logs or related services?

**Comment by SuXiChao1101** [comment]
When using the "adguardhome.exe - s run" command, it will prompt:cannot initialize syslog: The interface is unknown.It's really a bit embarrassing...

**Comment by iam-py-test** [comment]
Out of curiosity, why have you disabled Windows event logging?

**Reply by SuXiChao1101** [reply]
Because it is said that disabling EventLog can reduce the amount of disk writes and slightly increase the lifespan of the hard drive. Although I also know that this may not have much effect......

**Comment by SuXiChao1101** [comment, marked as answer]
Just tested in the virtual machine. Simply change the value of "Start" on the right side of [HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\WMI\Autologger\EventLog-System] to 1.

However, after looking at ADGuardHome and other software under [HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Services\EventLog\Application], I found that ADGuardHome calls %SystemRoot%\System32\EventCreate.exe to record logs. In fact, I think calling its own VMware\vmeventmsg.dll like VMWare is also a very good solution. I wonder if ADGuardHome will make such adjustments in the future...
